include/xtensor/xcontainer.hpp

```diff
@@ -220,7 +190,7 @@ namespace xt
         using inner_strides_type = typename base_type::inner_strides_type;
         using inner_backstrides_type = typename base_type::inner_backstrides_type;
 
-        void reshape(const shape_type& shape);
+        void reshape(const shape_type& shape, bool force = false);
         void reshape(const shape_type& shape, layout l);
         void reshape(const shape_type& shape, const strides_type& strides);
```

Comment:
The last two overloads should be available for dynamic layout only. Else you may end up with a container whose strides are not consistent with the layout.